codec-memcache/src/test/java/io/netty/handler/codec/memcache/binary/BinaryMemcacheObjectAggregatorTest.java

```diff
@@ -73,4 +77,30 @@ public void shouldAggregateChunksOnDecode() {
 
         channel.finish();
     }
+
+    @Test
+    public void shouldRetainByteBufWhenAggregating() {
+        channel = new EmbeddedChannel(
+                new BinaryMemcacheRequestEncoder(),
+                new BinaryMemcacheRequestDecoder(),
+                new BinaryMemcacheObjectAggregator(MAX_CONTENT_SIZE));
+
+        DefaultMemcacheContent content1 =
+                new DefaultMemcacheContent(Unpooled.copiedBuffer("Netty", CharsetUtil.UTF_8));
+        DefaultLastMemcacheContent content2 =
+                new DefaultLastMemcacheContent(Unpooled.copiedBuffer(" Rocks!", CharsetUtil.UTF_8));
+        int totalBodyLength = content1.content().readableBytes() + content2.content().readableBytes();
+
+        String key = "Netty";
+        BinaryMemcacheRequest request = new DefaultBinaryMemcacheRequest(key, Unpooled.buffer(8));
```

Comment:
do we want to put something in the extras and check that these are correct after aggregating ?